### PR TITLE
doc: move development process from the wiki

### PR DESCRIPTION
**Pull request by nashif**
move development model from the wiki to the main documentation to keep
it in sync and updated all the time and to get proper reviews.

### CODE REVIEW

**Comment by nashif** [comment]
DNM, still needs a few figures

File: doc/development_process/index.rst [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+project to provide users with a balance of the latest technologies and features
+and excellent overall quality. A roughly 3-month release cycle allows the project to
+coordinate development of the features that have actually been implemented,
+allowing the project to maintain the quality of the overall release without delays
+because of one or two features that are not ready yet.
+
+The Zephyr release model is loosely based on the Linux kernel model:
+
+- Release tagging procedure:
+  - linear mode on master,
```
Comment (dbkinder) [inline doc/development_process/index.rst]: add a blank line before the first item of a list

File: doc/development_process/index.rst [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  - release branches for maintenance after release tagging.
+- Each release period will consist of a merge window period followed by one or
+  more release candidates on which only stabilization changes, bug fixes, and
+  documentation can be merged in. - Merge window mode: all changes are accepted
```
Comment (dbkinder) [inline doc/development_process/index.rst]: ``- Merge window mode...`` is the beginning of a second-level list

File: doc/development_process/index.rst [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+  (subject to approval from the respective maintainers.) - When the merge window
```
Comment (dbkinder) [inline doc/development_process/index.rst]: ``- When the merge...`` is the second sub-bullet

File: doc/development_process/index.rst [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+  - CI sees the tag, builds and runs tests; QA analyses the report from the
+    build and test run and gives an ACK/NAK to the build
+  - Gatekeeper, with QA and any other needed input, determines if the release
+    candidate is a go for release
+  - If it is a go for a release, gatekeeper lays a tag release vN at the same
+    point
+- Development on new features continues in feature branches. Once features are
+  ready, they are submitted to mainline during the merge window period and after
+  the release is tagged.
+
```
Comment (dbkinder) [inline doc/development_process/index.rst]: picture from the wiki is missing here...

File: doc/development_process/index.rst [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+other in-tree code, they cannot cause regressions and should be safe to add at
+any time).
+
+As fixes make their way into the mainline, the patch rate will slow over time.
+The mainline gatekeeper releases new -rc drops once or twice a week; a normal
+series will get up to somewhere between -rc4 and -rc6 before the code base is
+considered to be sufficiently stable and the final 0.4.x release is made.
+
+At that point, the whole process starts over again.
+
```
Comment (dbkinder) [inline doc/development_process/index.rst]: Picture from the wiki missing here

File: doc/development_process/index.rst [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+Here is the description of the various moderation levels:
+
+- Low:
+  - Major New Features
```
Comment (dbkinder) [inline doc/development_process/index.rst]: need a blank line before the start of this sublist

File: doc/development_process/index.rst [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+  - Bug Fixes
+- Refactoring
+  - Structure/Directory Changes
```
Comment (dbkinder) [inline doc/development_process/index.rst]: need a blank line before the start of this sublist (more cases below)

File: doc/development_process/index.rst [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+***********************
+
+Long-term support releases are designed to be supported for a longer than normal
+period and will be the basis for products and certification for various usages.
+
+The LTS is released every 2 years.
+
+An LTS release will be branched and maintained independently of the mainline
+tree.
+
```
Comment (dbkinder) [inline doc/development_process/index.rst]: Picture in the wiki is missing here.

File: doc/development_process/index.rst [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+GitHub is intended to provide a framework for reviewing every commit before it
+is accepted into the code base. Changes, in the form of Pull Requests (PR) are
+uploaded to GitHub but don’t actually become a part of the project until they’ve
+been reviewed, passed a series of checks (CI) and approved by maintainers.
+GitHub is used to support the standard open source practice of submitting
+patches, which are then reviewed by the project members before being applied to
+the code base.
+
+The Zephyr project uses GitHub for code reviews and GIT tree management. When
+submitting a change or an enhancement to any Zephyr component developer should
```
Comment (dbkinder) [inline doc/development_process/index.rst]:
```suggestion
submitting a change or an enhancement to any Zephyr component, a developer must
```

File: doc/development_process/index.rst [hunk 10]
```diff
@@ -0,0 +1,286 @@
...
+Any failures found during the CI test run will result in a negative review.
+Developers are expected to fix issues and rework their patches and submit again.
+
+The CI infrastructure currently runs the following tests:
+
+- Run '''checkpatch''' for code style issues (can vote -1 on errors)
+- Gitlint: Git commit style based on project requorements
+- License Check: Check for conflicting licenses
+- Run '''sanitycheck''' script:
+  - Run kernel tests in QEMU (can vote -1 on errors)
```
Comment (dbkinder) [inline doc/development_process/index.rst]: add blank line before this sublist

**Review by dbkinder — DISMISSED** [review]

File: doc/development_process/index.rst [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+- LTS: Long Term Support
+
+Release Cycle
+*************
+
+The Zephyr project releases on a time-based cycle, rather than a feature-driven
+one. Zephyr releases represent an aggregation of the work of many contributors,
+companies, and individuals from the community.
+
+A time-based release process will enable the Zephyr
```
Comment (aescolar) [inline doc/development_process/index.rst]: Would it be better in present? "A time-based release process enables"

File: doc/development_process/index.rst [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+
+  - linear mode on master,
+  - release branches for maintenance after release tagging.
+- Each release period will consist of a merge window period followed by one or
+  more release candidates on which only stabilization changes, bug fixes, and
+  documentation can be merged in.
+
+  - Merge window mode: all changes are accepted (subject to approval from the
+    respective maintainers.)
+  - When the merge window is closed, gatekeeper lays a vN-rc1 tag and the tree
```
Comment (aescolar) [inline doc/development_process/index.rst]: is closed, +the+ gatekeeper (?)
(also twice in the next 3 paragraphs)

File: doc/development_process/index.rst [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+of a representative number of boards and platforms (subject to hardware
+availability in the CI infrastructure). Documentation changes are also verified
+through review and build testing to verify doc generation will be successful.
+
+Any failures found during the CI test run will result in a negative review.
+Developers are expected to fix issues and rework their patches and submit again.
+
+The CI infrastructure currently runs the following tests:
+
+- Run '''checkpatch''' for code style issues (can vote -1 on errors)
```
Comment (aescolar) [inline doc/development_process/index.rst]: Silly comment: This list is a bit out of date, would be nice to update it eventually

File: doc/development_process/index.rst [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+  - Run kernel tests in QEMU (can vote -1 on errors)
+  - Build various samples for different boards (can vote -1 on errors)
+
+
+Dealing with Proposals and RFCs
+*******************************
+
+Feature or enhancement proposals for Zephyr are used to propose a new design or
+modify an existing feature and get it approved. Currently, the developer mailing
+list is used to make such proposals. To better integrate with the current tools
```
Comment (aescolar) [inline doc/development_process/index.rst]: "Currently, the developer mailing list is used to make such proposals"
I guess this is not anymore the case?

File: doc/development_process/index.rst [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+- Open an Improvement GitHub Issue with an abstract and the proposal itself
+- Send an RFC with the abstract to the mailing list with an abstract and the
+  link to the detailed proposal
+- Approve/Reject/Defer based on feedback and comments
+
+Bug Reporting and Feature Tracking
+***********************************
+
+To maintain traceability and relation between proposals, changes, features, and
+issues, cross-referencing a commit with a GitHub Issue and vice versa should be
```
Comment (aescolar) [inline doc/development_process/index.rst]: Sidetrack: Shouldn't (we) reviewers actually enforce this?

File: doc/development_process/index.rst [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+communication tool by project member, contributors, and the community. The
+mailing list is open for topics related to the project and should be used for
+collaboration among team members working on the same feature or subsystem or for
+discussion project direction and daily development of the code base. In general,
+bug reports and issues should be entered and tracked in the bug tracking system
+(GitHub Issues) and not broadcasted to the mailing list, the same applies to
+code reviews. Code should be submitted to GitHub using the appropriate tools.
+
+To keep the community involved and informed summaries of GitHub Issues and PR
+traffic are posted to the mailing list on daily basis.
```
Comment (aescolar) [inline doc/development_process/index.rst]: not really, or?

**Review by aescolar — COMMENTED** [review]
Nice idea.
Just a couple of comments about the content itself in case you would feel like updating it.

File: doc/development_process/index.rst (hunk 16 above)
Comment (nashif) [reply doc/development_process/index.rst]: removed

File: doc/development_process/index.rst (hunk 13 above)
Comment (nashif) [reply doc/development_process/index.rst]: yes, will be updated with more details. want to create a good enough baseline to move away from the wiki first.

File: doc/development_process/index.rst (hunk 15 above)
Comment (nashif) [reply doc/development_process/index.rst]: we should I think.

File: doc/development_process/index.rst [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+
+    Merge Window
+
+Here is the description of the various moderation levels:
+
+- Low:
+
+  - Major New Features
+  - Bug Fixes
+- Refactoring
```
Comment (aescolar) [inline doc/development_process/index.rst]: Shouldnt refactoring be indented in the same level as bugfixes and structure directory changes?

**Review by aescolar — APPROVED** [review]

**Pull request merged**